File: src/test/java/rzaevali/DateUtilsTest.java
package rzaevali;

import com.google.common.collect.ImmutableList;
import org.junit.Test;
import rzaevali.utils.DateUtils;

import java.time.LocalDate;
import java.time.ZoneId;
import java.util.List;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class DateUtilsTest {

    @Test
    public void testDate() {
        for (int month = 1; month <= 7; ++month) {
            assertTrue(DateUtils.isSpring(LocalDate.of(2017, month, 1)));
        }

        for (int month = 8; month <= 12; ++month) {
            assertTrue(DateUtils.isAutumn(LocalDate.of(2017, month, 1)));
        }
    }

    @Test
    public void testDateComparison() {
        List<String> firstDate = ImmutableList.of("11102017", "25102017");
        List<String> secondDate = ImmutableList.of("31102017", "13112017");

        assertTrue(DateUtils.compareDates(firstDate, secondDate) < 1);
    }

    @Test
    public void testDateParsing() {
        LocalDate original = LocalDate.of(2018, 1, 2);
        LocalDate parsed = DateUtils.parseDate("02012018");

        assertEquals(original, parsed);
    }

}

File: src/test/kotlin/rzaevali/unit/DateUtilsTest.kt
package rzaevali.unit

import org.junit.Assert.assertEquals
import org.junit.Assert.assertTrue
import org.junit.Test
import rzaevali.utils.compareDates
import rzaevali.utils.isAutumn
import rzaevali.utils.isSpring
import rzaevali.utils.toLocalDate
import java.time.LocalDate

class DateUtilsTest {

    @Test
    fun testDate() {
        for (month in 1..7) {
            assertTrue(isSpring(LocalDate.of(2017, month, 1)))
        }

        for (month in 8..12) {
            assertTrue(isAutumn(LocalDate.of(2017, month, 1)))
        }
    }

    @Test
    fun testDateComparison() {
        val firstDate = listOf("11102017", "25102017")
        val secondDate = listOf("31102017", "13112017")

        assertTrue(compareDates(firstDate, secondDate) < 1)
    }

    @Test
    fun testDateParsing() {
        val original = LocalDate.of(2018, 1, 2)
        val parsed = "02012018".toLocalDate()

        assertEquals(original, parsed)
    }

}
